### PR TITLE
@compileTimeOnly for macro annotations

### PR DESCRIPTION
**Pull request by xeno-by**

### CODE REVIEW

File: overviews/macros/annotations.md [hunk 1]
```diff
@@ -38,12 +38,20 @@ that the enclosing annotation is a macro annotation).
...
+    @compileTimeOnly("enable macro paradise to expand macro annotations")
     class identity extends StaticAnnotation {
       def macroTransform(annottees: Any*) = macro ???
     }
 
-The `macroTransform` macro is supposed to take a list of untyped annottees (in the signature their type is represented as `Any`
+First of all, note the `@compileTimeOnly` annotation. It is not mandatory, but is recommended to avoid confusion.
+Macro annotations look like normal annotations to the vanilla Scala compiler, so if you forget to enable the macro paradise
+plugin in your build, your annotations will silently fail to expand. The `@compileTimeOnly` annotation makes sure that
+no references to the underlying definition is present in the program code after typer, so it will prevent the aforementioned
```
Comment (andrewdbate) [inline overviews/macros/annotations.md]: Change "definition is present" to "definition are present".

Comment (andrewdbate) [inline overviews/macros/annotations.md]: Or better still, to prefer the singular, change "sure that no references to the underlying definition is present" to "sure that no reference to the underlying definition is present" (delete the s from references).

Comment (xeno-by) [inline overviews/macros/annotations.md]: Done. Is the change otherwise LGTY?

Comment (andrewdbate) [inline overviews/macros/annotations.md]: Yes, that looks great!

Comment (xeno-by) [inline overviews/macros/annotations.md]: Thanks! I'll merge then.

**Pull request merged**